app/javascript/components/Editor.tsx

```diff
@@ -424,7 +424,10 @@ export function Editor({
           <RunTestsButton onClick={runTests} />
           <SubmitButton
             onClick={submit}
-            disabled={submission?.testRun?.status !== TestRunStatus.PASS}
+            disabled={
+              submission?.testRun?.status !== TestRunStatus.PASS ||
+              !isEqual(submissionFilesRef.current, files)
+            }
```

Comment:
Should we add a test to verify this behavior?

Reply:
Thanks. I've added the test.